test.py

```diff
@@ -1,124 +1,527 @@
+            self.contents = contents
+
+
+        def read(self):
+            return self.contents
+
+
+    def setUp(self):
+        self.travis = TravisCiApiProvider()
+
+
+    def test_get_build(self, urlopen_mock):
+        urlopen_mock.return_value = self.FakeFile(json.dumps({"Something":"here"}))
+        self.travis.get_build(1)
+        urlopen_mock.assert_called_with(TravisCiApiProvider.build_url.format(build_id=1))
+
+
+    def test_get_log(self, urlopen_mock):
+        job_id = 1
+        urlopen_mock.return_value = self.FakeFile(open('resources/single-line-comment.log').read())
+        self.travis.get_log({"matrix":[{'id':job_id}]})
+        urlopen_mock.assert_called_with(TravisCiApiProvider.log_url.format(job_id=job_id))
+
+
+    def test_get_pull_request_number(self, urlopen_mock):
+        pr_num = 1234
+        self.assertEquals(1234, self.travis.get_pull_request_number({"compare_url":"https://github.com/servo/servo/{}".format(pr_num)}))
+
+
+class TestErrorLogParser(unittest.TestCase):
+    def setUp(self):
+        self.log_parser = ErrorLogParser()
+
+
+    def test_parse_log(self):
+        with self.assertRaises(NotImplementedError):
+            self.log_parser.parse_log("log", "regex")
+
+
+class TestServoErrorLogParser(unittest.TestCase):
+    def setUp(self):
+        self.error_parser = ServoErrorLogParser()
+        self.multi_log = open('resources/multi-line-comment.log').read()
+        self.expected_multi_errors = error_sample
+        self.single_log = open('resources/single-line-comment.log').read()
+        self.expected_single_errors = [
+            {
+                'body': 'missing space before {', 
+                'position': 49, 
+                'path': './components/plugins/lints/sorter.rs'
+            }
+        ]
+
+
+    def test_parse_errors(self):
+        self.assertEqual(self.expected_multi_errors, list(self.error_parser.parse_log(self.multi_log)))
+        self.assertEqual(self.expected_single_errors, list(self.error_parser.parse_log(self.single_log)))
+
+
+class TestPayloadHandler(unittest.TestCase):
+    def setUp(self):
+        self.handler = PayloadHandler()
+
+
+    def test_handle_payload(self):
+        with self.assertRaises(NotImplementedError):
+            self.handler.handle_payload("payload")
+
+
+class TestTravisPayloadHandler(unittest.TestCase):    
+    def setUp(self):
+        class TravisDouble():
+            def get_build(self, build_id):
+                return 1
+
+
+            def get_log(self, build_data):
+                return open('resources/multi-line-comment.log').read()
+
+
+            def get_pull_request_number(self, build_data):
+                return 1
+
+
+        class ErrorParserDouble():
+            path_key = ServoErrorLogParser.path_key
+            body_key = ServoErrorLogParser.body_key
+            position_key = ServoErrorLogParser.position_key
+
+            def parse_log(self, log):
+                return error_sample
```

Comment:
I'd be tempted to put the contents of error_sample in a file, rather than having it inline in this one.